main/modes/demo/demoMode.c

```diff
@@ -324,7 +370,13 @@ static int16_t demoAdvancedUSB(uint8_t* buffer, uint16_t length, uint8_t isGet)
  */
 static void demoConCb(p2pInfo* p2p, connectionEvt_t evt)
 {
-    // Do something
+    if (evt == CON_ESTABLISHED)
+    {
+        const uint8_t testMsg[] = {0x01, 0x02, 0x03, 0x04};
+        p2pSendMsg(&dv->p2p, testMsg, ARRAY_SIZE(testMsg), demoMsgTxCbFn);
```

Comment:
This transmission gets the counters out of sync, at least on windows. If I comment it out, the counters stay in sync while tapping the button
![image](https://github.com/AEFeinstein/Swadge-IDF-5.0/assets/231180/a4db6e30-ce79-41db-a62c-36eaf5df6f32)
Trace with your new fancy tool:
```
0A:37:71:71:99:1D > P2P p  CONNECT
0A:37:71:71:99:1D > P2P p  CONNECT
0A:37:71:71:99:1D > P2P p  CONNECT
60:9F:BB:CC:99:A1 > P2P p  CONNECT
0A:37:71:71:99:1D > P2P p  START #000 > 60:9F:BB:CC:99:A1
0A:37:71:71:99:1D > P2P p  START #000 > 60:9F:BB:CC:99:A1
0A:37:71:71:99:1D > P2P p  START #000 > 60:9F:BB:CC:99:A1
60:9F:BB:CC:99:A1 > P2P p  ACK   #000 > 0A:37:71:71:99:1D
60:9F:BB:CC:99:A1 > P2P p  ACK   #001 > 0A:37:71:71:99:1D
60:9F:BB:CC:99:A1 > P2P p  ACK   #002 > 0A:37:71:71:99:1D
0A:37:71:71:99:1D > P2P p  CONNECT
60:9F:BB:CC:99:A1 > P2P p  START #003 > 0A:37:71:71:99:1D
60:9F:BB:CC:99:A1 > P2P p  START #003 > 0A:37:71:71:99:1D
0A:37:71:71:99:1D > P2P p  ACK   #001 > 60:9F:BB:CC:99:A1
0A:37:71:71:99:1D > P2P p  DATA  #002 > 60:9F:BB:CC:99:A1 [ 01 02 03 04 ]
0A:37:71:71:99:1D > P2P p  ACK   #003 > 60:9F:BB:CC:99:A1
0A:37:71:71:99:1D > P2P p  DATA  #002 > 60:9F:BB:CC:99:A1 [ 01 02 03 04 ]
60:9F:BB:CC:99:A1 > P2P p  DATA  #004 > 0A:37:71:71:99:1D [ 01 02 03 04 ]
60:9F:BB:CC:99:A1 > P2P p  ACK   #005 > 0A:37:71:71:99:1D
60:9F:BB:CC:99:A1 > P2P p  ACK   #006 > 0A:37:71:71:99:1D
0A:37:71:71:99:1D > P2P p  ACK   #004 > 60:9F:BB:CC:99:A1
```

Reply:
Ah, initially it was sending the first packet right after the initialize call, and it ended up doing very weird stuff - it would send a whole ton of data packets like this:

`0A:37:71:71:99:1D > P2P p DATA #000 > 00:00:00:00:00:00 [ 01 02 03 04 ]`

I'll just have the packets only on the button then, it seems like the first one isn't getting through since it's not quite initialized yet.

Reply:
I tested this on real hardware, and it definitely works! And also has the same issue! I think you've stumbled on a race condition. What's _supposed_ to happen is `p2pGetPlayOrder()` is checked and only the first player sends this packet. Let me see if that keeps everything nice.

Reply:
All good after 6915c8279162df8a1aa0cd897d8f208256de7325